src/main/java/core/basesyntax/serviceimpl/WriterServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.WriterService;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+public class WriterServiceImpl implements WriterService {
+
+    @Override
+    public void writeAll(Path pathToFile) {
+        String report = createReport();
+        try (BufferedWriter bufferedWriter = new BufferedWriter(
+                new FileWriter(new File(pathToFile.toUri())))) {
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Error occurred while writing to file" + pathToFile, e);
+        }
+    }
+
+    private String createReport() {
```

Comment:
It’s not appropriate logic for writer service. Let’s  create separate ReportService